### PR TITLE
Feature/slf4j with logback

### PR DESCRIPTION
**Pull request by kirvin**

### CODE REVIEW

**Comment by charlesweird** [comment]
@kirvin just curious what's the reason we are switching to slf4j? thanks

**Comment by kirvin** [comment]
I would like to standardize on SLF4J + logback, as these libraries are more active (logback in particular is considered best practice compared to Log4J).  SLF4J also gives us a nicer abstraction across multiple logging frameworks, in case we need to switch for some reason at a later time.

Here are some relevant links:
  * http://www.slf4j.org/index.html
  * http://logback.qos.ch/reasonsToSwitch.html

File: src/main/java/com/mellmo/roambi/cli/RoambiCmdLineClient.java [hunk 1]
```diff
@@ -78,7 +80,10 @@ public int getTotalCount() {
     @Override
     protected void doExecute(String cmd) throws Exception {
         if (verbose) {
-            Logger.getRootLogger().setLevel(Level.DEBUG);
+            //LoggerFactory.getLogger(Logger.ROOT_LOGGER_NAME).setLevel(Level.DEBUG);
```
Comment (charlesweird) [inline src/main/java/com/mellmo/roambi/cli/RoambiCmdLineClient.java]: the diff here (the before) doesn't match the one I checked in last week. 
```     protected void doExecute(String cmd) throws Exception {
         if (verbose) {
             Logger.getRootLogger().setLevel(Level.DEBUG);
             Logger.getLogger("httpclient").setLevel(Level.DEBUG);
             Logger.getLogger("org.apache.commons.httpclient").setLevel(Level.DEBUG);
         }
         if (cmd == null) {
```
???

**Comment by charlesweird** [comment]
so SLF4J does not have his own version of log4j.properties? we can get rid of log4j.properties once for all?

**Comment by kirvin** [comment]
@charlesweird Updated `RoambiCmdLineClient` to try and set all levels explicitly.  The SLF4J api does not have this functionality, so we have to do some casting, but this seems to be the accepted practice for this kind of functionality.

**Comment by kirvin** [comment]
Unfortunately, this doesn't mean we get to eliminate config files; we just change `log4j.properties` for `logback.xml`...

http://logback.qos.ch/manual/configuration.html

File: src/main/java/com/mellmo/roambi/cli/RoambiCmdLineClient.java [hunk 2]
```diff
@@ -78,9 +80,10 @@ public int getTotalCount() {
     @Override
     protected void doExecute(String cmd) throws Exception {
         if (verbose) {
-            Logger.getRootLogger().setLevel(Level.DEBUG);
-            Logger.getLogger("httpclient").setLevel(Level.DEBUG);
-            Logger.getLogger("org.apache.commons.httpclient").setLevel(Level.DEBUG);
+        	// this ties us directly to the logback implementation...
+        	((ch.qos.logback.classic.Logger) LoggerFactory.getLogger(Logger.ROOT_LOGGER_NAME)).setLevel(Level.DEBUG);
+        	((ch.qos.logback.classic.Logger) LoggerFactory.getLogger("httpclient")).setLevel(Level.DEBUG);
+        	((ch.qos.logback.classic.Logger) LoggerFactory.getLogger("org.apache.commons.httpclient")).setLevel(Level.DEBUG);
```
Comment (charlesweird) [inline src/main/java/com/mellmo/roambi/cli/RoambiCmdLineClient.java]: so previous we had to set httpclient's log level here in order to overwrite its setting in log4j.properties. If slf4j does not have a configuration properties file, then shouldn't setting the RootLogger automatically apply to all the classes regardless?

Comment (charlesweird) [inline src/main/java/com/mellmo/roambi/cli/RoambiCmdLineClient.java]: @kirvin just got your last comment, please ignore my last comment/questions thanks

Comment (charlesweird) [inline src/main/java/com/mellmo/roambi/cli/RoambiCmdLineClient.java]: @kirvin are we going to include a default logback.xml in the jar?

**Comment by kirvin** [comment]
I think this is ready for review now.  @charlesweird let me know if there is anything else that needs to change.

**Comment by charlesweird** [comment]
Out of curiosity I was looking at the javadoc to trying to figure out slf4j's official way to set logger level but I can't seem to find it. But your way is just fine.

**Comment by kirvin** [comment]
Thanks!  Could you look at https://github.com/Roambi/roambi-java-sdk/pull/10 also when you have a chance?

**Pull request merged**